### ISSUE TITLE
problem importing h5py after importing taurus

### ISSUE THREAD

**Issue by cpascual** [piertoni/taurus#15]
In the current futurize1 branch, using the taurus-test-py3 docker image, I am getting trouble importing the `extra_nexus` submodule:

```
root@taurus-test:/# python -m taurus.qt.qtgui.extra_nexus
MainThread     INFO     2018-10-15 08:53:35,957 TaurusRootLogger: Using "PyQt4" for Qt
MainThread     INFO     2018-10-15 08:53:36,076 taurus.qt.qtgui.icon.icon: Setting Tango icon theme (from /usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/icon/)
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 163, in _run_module_as_main
    mod_name, _Error)
  File "/usr/lib/python2.7/runpy.py", line 111, in _get_module_details
    __import__(mod_name)  # Do not catch exceptions initializing package
  File "/usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/extra_nexus/__init__.py", line 30, in <module>
    from .taurusnexuswidget import TaurusNeXusBrowser
  File "/usr/local/lib/python2.7/dist-packages/taurus/qt/qtgui/extra_nexus/taurusnexuswidget.py", line 36, in <module>
    from PyMca5.PyMcaGui.io.hdf5 import HDF5Widget, HDF5Info, HDF5DatasetTable
  File "/usr/lib/python2.7/dist-packages/PyMca5/PyMcaGui/io/hdf5/HDF5Widget.py", line 46, in <module>
    import h5py
  File "/usr/lib/python2.7/dist-packages/h5py/__init__.py", line 60, in <module>
    from .tests import run_tests
  File "/usr/lib/python2.7/dist-packages/h5py/tests/__init__.py", line 15, in <module>
    from . import old, hl
  File "/usr/lib/python2.7/dist-packages/h5py/tests/old/__init__.py", line 4, in <module>
    from . import ( test_attrs,
  File "/usr/lib/python2.7/dist-packages/h5py/tests/old/test_group.py", line 41, in <module>
    fsencode(u"α")
  File "/usr/lib/python2.7/dist-packages/h5py/_hl/compat.py", line 68, in fsencode
    return filename.encode(encoding, errors)
  File "/usr/lib/python2.7/dist-packages/future/utils/surrogateescape.py", line 59, in surrogateescape_handler
    decoded = replace_surrogate_encode(mystring)
  File "/usr/lib/python2.7/dist-packages/future/utils/surrogateescape.py", line 86, in replace_surrogate_encode
    raise exc
NameError: global name 'exc' is not defined

```

Oddly, in my own machine (with the same taurus and h5py versions installed, I do not get the same problem!!)

Anyway: I managed to reproduce it in the docker without taurus:
```python
from future.utils.surrogateescape import register_surrogateescape
register_surrogateescape()
import h5py
```

<s>... but I cannot find where (if so) in taurus we are registering the surrogateescape</s>
UPDATE: ...and while surrogateescape is not directly registered in taurus, it is done indirectly when doing

```python
from future import standard_library
standard_library.install_aliases()
```

Which is done in several points of the futurize branch.

**Comment by cpascual** [comment]
I am not sure on how to solve this one. It is a bug in `future.utils.surrogateescape`:

https://github.com/PythonCharmers/python-future/issues/256

This only affects nexus widget AFAIK and only on some machines (e.g. in my machine I am not affected, just in the docker...), so I propose to wait for it to be solved in the `future` module on its own (and just create an issue on Taurus to keep a trace on this)

**Comment by cpascual** [comment]
A ticket was created by @reszelaz in taurus : https://github.com/taurus-org/taurus/issues/836 
... and we closed it since it is [already fixed with h5py 2.8](https://github.com/taurus-org/taurus/issues/836#issuecomment-451409412). 

Consequently, I am also closing this one.
